docs/reference/search/search-your-data/learning-to-rank-model-training.asciidoc

```diff
@@ -0,0 +1,166 @@
+    QueryFeatureExtractor(
+        feature_name="title_bm25",
+        query={"match": {"title": "{{query}}"}}
+    ),
+    # We can use a script_score query to get the value of the field rating directly as a feature:
+    QueryFeatureExtractor(
+        feature_name="popularity",
+        query={
+            "script_score": {
+                "query": {"exists": {"field": "popularity"}},
+                "script": {"source": "return doc['popularity'].value;"},
+            }
+        },
+    ),
+    # We can execute a script on the value of the query and use the return value as a feature:
+    QueryFeatureExtractor(
+        feature_name="query_length",
+        query={
+            "script_score": {
+                "query": {"match_all": {}},
+                "script": {
+                    "source": "return params['query'].splitOnToken(' ').length;",
+                    "params": {
+                        "query": "{{query}}",
+                    }
+                },
+            }
+        },
+    ),
+]
+----
+// NOTCONSOLE
+
+Once the feature extractors have been defined, they are wrapped in an `eland.ml.ltr.LTRModelConfig` object for use in later training steps:
+
+[source,python]
+----
+from eland.ml.ltr import LTRModelConfig
+
+ltr_config = LTRModelConfig(feature_extractors)
+----
+// NOTCONSOLE
+
+[discrete]
+[[learning-to-rank-model-training-feature-extraction]]
+===== Extracting features for training
+
+Building your dataset is a critical step in the training process. This involves
+extracting relevant features and adding them to your judgment list. We
+recommend using Eland's `eland.ml.ltr.FeatureLogger` helper class for this
+process.
+
+[source,python]
+----
+from eland.ml.ltr import FeatureLogger
+
+# Create a feature logger that will be used to query {es} to retrieve the features:
+feature_logger = FeatureLogger(es_client, MOVIE_INDEX, ltr_config)
+----
+// NOTCONSOLE
+
+The FeatureLogger provides an `extract_features` method which enables you to extract features for a list of specific documents from your judgment list. At the same time, you can pass query parameters to the feature extractors defined earlier:
+
+[source,python]
+----
+feature_logger.extract_features(
+    query_params:{"query": "foo"},
+    doc_ids=["doc-1", "doc-2"]
+)
+----
+// NOTCONSOLE
+
+Our https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/08-learning-to-rank.ipynb[example notebook^] explains how to use the `FeatureLogger` to build a training dataset, by adding features to a judgment list.
+
+[discrete]
+[[learning-to-rank-model-training-feature-extraction-notes]]
+====== Notes on feature extraction
+
+* We strongly advise against implementing feature extraction on your own. It's crucial to maintain consistency in feature extraction between the training environment and inference in {es}. By using eland tooling, which is developed and tested in tandem with {es}, you can ensure that they function together consistently.
```

Comment:
Should this be a warning?

Reply:
The warning presentation does not fit well with the other notes. I prefer to keep this into a note.